### ISSUE TITLE
Research ArchivesSnake abstraction layer functionality

### ISSUE THREAD

**Issue by bonniegee** [RockefellerArchiveCenter/rac_aspace#2]
Saving and deleting in ArchivesSpace is not covered by the abstraction layer of ArchivesSnake. Reach out to ArchivesSnake team for background on why.

**Comment by p-galligan** [comment]
Spoke a little about this with Dave Mayo. The reason it's edit-only right now is twofold: manpower/time was limited and partly because the team was unsure about a "safe" editing method generalizing across all of the different JSONModel types in AS.

It _could_ be extended to do edits, but, for example, the process of editing an archival object so it's linked to a digital object or has a new digital object, is several interaction across several objects. But for the abstraction layer to make conceptual sense, it should only be a single interaction. 

Treating the abstraction layer as an ORM should be feasible, but it would either need to be implemented in ASnake itself or ASnake would have to grow a real plugin/extension mechanism, which Dave thinks is unfeasible given the manpower/time it would take.

**Comment by p-galligan** [comment]
Dave says he has a beginning of a proof of concept somewhere, trying to convince him to share it if he's comfortable.

**Comment by p-galligan** [comment]
https://github.com/archivesspace-labs/ArchivesSnake/tree/save_functionality

Rebased back onto master, but it should work for simple things like:

`repo = aspace.repository(2) 
 repo.name = "changed" 
 repo.save()`

**Comment by helrond** [comment]
@p-galligan can you summarize where in the codebase these changes are?

**Comment by p-galligan** [comment]
I can summarize where the current changes are in the proof of concept, but I'm not sure where everything would have to change to fully implement edit/save functionality. I get the sense that those changes would be pretty large.

Currently the changes in save_functionality exist almost entirely in the `asnake/jsonmodel/__init__.py` file. Generally between lines 100-300. There are some changes to the tests as well, but those are smaller.

**Comment by helrond** [comment]
Linking to the diff for quick reference: https://github.com/archivesspace-labs/ArchivesSnake/compare/save_functionality?expand=1#diff-44fe0efbfc846c3189d2724f8762f368
